### ISSUE TITLE
[gcdserver] Fill I3OMGeo.orientation with DOM tilt and rotation information (Trac #2074)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2066]
A feature of the PPC should be implemented to clsim to use the feature for the simulation production.

Migrated from https://code.icecube.wisc.edu/ticket/2074
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:08", 
    "description": "A feature of the PPC should be implemented to clsim to use the feature for the simulation production.", 
    "reporter": "mase", 
    "cc": "mase, summer.blot, cweaver, dima", 
    "resolution": "fixed", 
    "_ts": "1550067308113782", 
    "component": "jeb + pnf", 
    "summary": "[gcdserver] Fill I3OMGeo.orientation with DOM tilt and rotation information", 
    "priority": "major", 
    "keywords": "", 
    "time": "2017-08-03T21:25:29", 
    "milestone": "", 
    "owner": "jbraun", 
    "type": "task"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170818T15:24:27: 
* david.schultz changed milestone from "" to "IceSim 6"
* david.schultz changed owner from "" to "claudio.kopper"
* david.schultz changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20171018T09:29:54: 
* jvansanten changed owner from "claudio.kopper" to "jbraun"
* jvansanten changed _comment0 from "clsim already uses `I3OMGeo.orientation` to calculate angles w.r.t. the PMT axis. The field just needs to be filled with a non-default value when creating GCD files for simulation (and for data, for that matter)." to "1508319989747654"
* jvansanten commented:

> clsim already uses `I3OMGeo.orientation` to calculate angles w.r.t. the PMT axis. The field just needs to be filled with a non-default value when creating GCD files for simulation (and for data, for that matter).
> 
> The current best fit values and uncertainties are in ppc/resources/ice/cx.dat (components of PMT axis) and ppc/resources/ice/dx.dat (azimuth of direction of the cable, which could be converted into a rotation around the PMT axis).


* jvansanten changed summary from "Implementation of the PPC feature to clsim (DOM tilt)" to "[gcdserver] Fill I3OMGeo.orientation with DOM tilt and rotation information"
* jvansanten changed component from "simulation" to "jeb + pnf"

**Comment by icecube-trac** [comment]
Trac update at 20171018T09:51:37: jvansanten commented:

> A temporary fix would be to modify I3GeometryDecomposer (or wrap it in a segment that) reads cx.dat and dx.dat and overwrites the orientation field read from the GCD file. This would allow some flexibility when the best-fit values are updated, but like all temporary fixes, risks becoming permanent.

**Comment by icecube-trac** [comment]
Trac update at 20171018T15:01:48: jbraun commented:

> gcdserver already handles DOM orientations, although at this point only the enumerated UP and DOWN are supported.  Extending this to arbitrary orientations is trivial, and putting the support for this into gcdserver is definitely the way to go.  A few points:
> 
> 1. gcdserver is able to support the latest, greatest, blessed tilt values and any previous, not-so-great tilt values.  It can't support tweaking the values e.g. for custom simulations, but that can be done easily with scripts.
> 
> 2. L2 processing will begin using these values as soon as they're inserted into the database and IceRec is patched with the gcdserver changes.  The plan needs to be presented/blessed on a Tuesday call before we can roll this out.  We can also push this to pole with a PnF release.
> 
> 3. If we're planning to update the tilt values more than once, then the database update procedure needs to be robust.  The values need to be stored in a file where the meaning is obvious.  Someone just looking at cx.dat has no idea what the data is, although columns 1 and 2 are obvious.  JSON is strongly preferred, but XML is OK.

**Comment by icecube-trac** [comment]
Trac update at 20171018T15:18:13: jbraun commented:

> Another issue to resolve: I3Geometry currently does not support cable azimuth values.  gcdserver could provide an OMKey --> azimuth map in the G frame instead, but maybe there is a better solution.  Some input is needed from those that support dataclasses.

**Comment by icecube-trac** [comment]
Trac update at 20171018T15:35:50: 
* jvansanten commented:

> If I understand correctly, the cable position is currently assumed to be fixed w.r.t. the flasher board, and so is degenerate with the rotation of the DOM. If that's true, then I think it's reasonable to stick with only an orientation field for now, and add an explicit cable position if and when we ever gain the ability to measure it independently.


* jvansanten changed _comment0 from "I think it's reasonable to omit an explicit cable azimuth for now, and assume that it is at a fixed angle w.r.t. the flasher board." to "1508341774417846"

**Comment by icecube-trac** [comment]
Trac update at 20171212T17:41:11: 
* jbraun commented:

> In changeset 160053:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="160053"
> Add support for arbitrary DOM orientations.  Closes #2066.
> ```


* jbraun changed status from "assigned" to "closed"
* jbraun changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20171212T17:43:55: jbraun commented:

> We still need final values from Dima and agreement to use these DOM tilt values in reconstruction.

**Comment by icecube-trac** [comment]
Trac update at 20171212T17:47:02: jbraun commented:

> Note also that the DOM orientation is defined with LED7 on the x-axis.  The cable angle is (DOM x-axis angle + 90 degrees) in detector coordinates.

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:08: 
* olivas changed milestone from "IceSim 6" to ""
* olivas commented:

> Milestone IceSim 6 deleted
